core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -369,6 +369,36 @@ class AdminClient(val time: Time,
     (response.error, response.tokens().asScala.toList)
   }
 
+  def deleteConsumerGroups(groups: List[String]): Map[String, Errors] = {
+    var errors: Map[String, Errors] = Map()
+    val groupsPerCoordinator = groups.map { group =>
```

Comment:
I think I'd suggest moving coordinator lookup to a separate function. You might also consider using the `Either` class to distinguish errors since the mixture of functional logic and updates to the mutable `errors` collection is a little odd.